### DISCUSSION TITLE
How to do assertions with the readline module?

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#18008] [Questions and Answers]
Hello. I have an issue with the `readline` module. I have this code
```
module main

import readline

fn main() {
     mut r := readline.Readline{}
     mut q1 := r.read_line('What's the name of creator of V? ')!
     a1 := "Alex Medvednikov"
     if q1 == a1 {
         println('You're correct")
     }
}
```
But the problem is that it doesn't work. I'm using V 0.3.3 on NixOS.

**Comment by i582** [comment]
Hi
Can you tell me what exactly is not working? Does it work in a regular terminal or does it not work in an IDEA terminal?

**Reply by fhaun** [reply]
Use `q1 = q1.trim_space()` to get rid of the newline.

**Comment by jlmess77** [comment]
It doesn't work in the playground or wasm as both will panic but it works fine in the normal terminal with a bit of tweaks. You need to add a backsplash before a single quote and follow @fhaun's suggestion. 

```
module main

import readline

fn main() {
     mut r := readline.Readline{}
     mut q1 := r.read_line('What\'s the name of creator of V? ')!
     a1 := "Alex Medvednikov"
     if q1.trim_space() == a1 {
         println('You\'re correct')
     } else {
         println('You\'re incorrect')
     }
}
```
